cpu/cc26x0/periph/timer.c

```diff
@@ -82,84 +133,120 @@ int timer_set(tim_t tim, int channel, unsigned int timeout)
 
 int timer_set_absolute(tim_t tim, int channel, unsigned int value)
 {
-    if (channel != 0) {
+    DEBUG("timer_set_absolute(%u, %u, %u)\n", tim, channel, value);
+
+    if ((tim >= TIMER_NUMOF) || (channel >= timer_config[tim].chn)) {
         return -1;
     }
-
-    dev(tim)->TAMATCHR = value;
-    dev(tim)->IMR |= GPT_IMR_TAMIM;
-
-    return 0;
+    switch (channel) {
+        case 0:
+            dev(tim)->ICLR = GPT_IMR_TAMIM;
+            dev(tim)->TAMATCHR = (timer_config[tim].cfg == GPT_CFG_32T) ?
+                                 value : (LOAD_VALUE - value);
+            dev(tim)->IMR |= GPT_IMR_TAMIM;
+            break;
+        case 1:
+            dev(tim)->ICLR = GPT_IMR_TBMIM;
+            dev(tim)->TBMATCHR = (timer_config[tim].cfg == GPT_CFG_32T) ?
+                                 value : (LOAD_VALUE - value);
+            dev(tim)->IMR |= GPT_IMR_TBMIM;
+            break;
+    }
+    return 1;
 }
 
 int timer_clear(tim_t tim, int channel)
 {
-    if (channel != 0) {
+    if ((tim >= TIMER_NUMOF) || (channel >= timer_config[tim].chn)) {
         return -1;
     }
 
-    dev(tim)->IMR &= ~(GPT_IMR_TAMIM);
+    dev(tim)->IMR &= (channel == 0) ? ~(GPT_IMR_TAMIM) : ~(GPT_IMR_TBMIM);
 
     return 0;
 }
 
 unsigned int timer_read(tim_t tim)
 {
-    return dev(tim)->TAV & 0xFFFF;
+    if (tim >= TIMER_NUMOF) {
+        return 0;
+    }
+    if (timer_config[tim].cfg == GPT_CFG_32T) {
+        return dev(tim)->TAV;
+    }
+    return LOAD_VALUE - (dev(tim)->TAV & 0xFFFF);
 }
 
 void timer_stop(tim_t tim)
 {
-    dev(tim)->CTL = 0;
+    if (tim < TIMER_NUMOF) {
+        dev(tim)->CTL = 0;
+    }
 }
 
 void timer_start(tim_t tim)
 {
-    dev(tim)->CTL = GPT_CTL_TAEN;
+    if (tim < TIMER_NUMOF) {
+        switch (timer_config[tim].chn) {
+            case 1:
+                dev(tim)->CTL = GPT_CTL_TAEN;
+                break;
+            case 2:
+                dev(tim)->CTL = GPT_CTL_TAEN | GPT_CTL_TBEN;
+                break;
+        }
+    }
 }
 
 /**
- * @brief           handle interrupt for a timer
+ * @brief   handle interrupt for a timer channel 0 (A)
  *
  * @param[in] tim   index of the timer
  */
-static inline void isr_handler(tim_t tim)
-{
-    uint32_t mis = dev(tim)->MIS;
+static void irq_handler_a(tim_t tim) {
+    uint32_t mis;
+    /* Latch the active interrupt flags */
+    mis = dev(tim)->MIS & TIMER_A_IRQ_MASK;
+    /* Clear the latched interrupt flags */
     dev(tim)->ICLR = mis;
 
     if (mis & GPT_IMR_TAMIM) {
+        /* Disable further match interrupts for this timer/channel */
         dev(tim)->IMR &= ~GPT_IMR_TAMIM;
+        /* Invoke the callback function */
         ctx[tim].cb(ctx[tim].arg, 0);
     }
 
     cortexm_isr_end();
 }
 
-#ifdef TIMER_0_ISR
-void TIMER_0_ISR(void)
-{
-    isr_handler(0);
-}
-#endif
+/**
```

Comment:
Have you thought about having a channel_config with *GPT_IMR_T[AB]MIM* and *TIMER_[AB]_IRQ_MASK*? And then merge the two ISRs to a ``irq_handler( tim_t tim, int channel)``?